selfdrive/test/process_replay/process_replay.py

```diff
@@ -186,6 +191,48 @@ def torqued_rcv_callback(msg, cfg, frame):
   return (frame - 1) == 0 or msg.which() == 'liveLocationKalman'
 
 
+def dmonitoringmodeld_rcv_callback(msg, cfg, frame):
+  return msg.which() == "driverCameraState"
+
+
+class ModeldCameraSyncRcvCallback:
+  def __init__(self):
+    self.road_present = False
+    self.wide_road_present = False
+    self.is_dual_camera = True
+
+  def __call__(self, msg, cfg, frame):
```

Comment:
I find functions written like this very confusing, there's a lot of returns and goes back and forth between state management and returning.

I would write it like this:

```
class ModeldCameraSyncRcvCallback:
  def __init__(self):
    self.road_present = False
    self.wide_road_present = False
    self.is_dual_camera = True

  def __call__(self, msg, cfg, frame):
    if msg.which() == "initData":
      self.is_dual_camera = msg.initData.deviceType in ["tici", "tizi"]
    elif msg.which() == "roadCameraState":
      self.road_present = True
    elif msg.which() == "wideRoadCameraState":
      self.wide_road_present = True

    if self.road_present and self.wide_road_present
      self.road_present, self.wide_road_present = False, False
      return True
    elif self.road_present and not self.dual_camera:
      self.road_present = False
      return True
    else:
      return False
```

Reply:
it was probably refactored N times, before ended up looking like this... your snippet looks much cleaner :+1: